src/node_crypto.cc

```diff
@@ -5586,15 +5585,15 @@ void PBKDF2(const FunctionCallbackInfo<Value>& args) {
 
   obj = env->pbkdf2_constructor_template()->
       NewInstance(env->context()).ToLocalChecked();
-  req = new PBKDF2Request(env,
-                          obj,
-                          digest,
-                          passlen,
-                          pass,
-                          saltlen,
-                          salt,
-                          iter,
-                          keylen);
+  std::unique_ptr<PBKDF2Request> req(new PBKDF2Request(env,
+                                                       obj,
+                                                       digest,
+                                                       passlen,
+                                                       pass,
+                                                       saltlen,
+                                                       salt,
+                                                       iter,
+                                                       keylen));
```

Comment:
Since you're here, I'd be okay with writing this as:
```c++
std::unique_ptr<PBKDF2Request> req(
    new PBKDF2Request(env, obj, digest, passlen, pass, saltlen, iter, keylen));
```